packages/wallet/test/hardware/ledger/LedgerKeyAgent.test.ts

```diff
@@ -298,6 +306,258 @@ describe('LedgerKeyAgent', () => {
+              anchor: null,
+              coldCredential: credential
+            }
+          ];
+          const tx = txBuilder
+            .addOutput(
+              txBuilder.buildOutput().address(outputs[0].address).coin(BigInt(outputs[0].value.coins)).toTxOut()
+            )
+            .build();
+
+          // expect(await tx.sign()).toBeTruthy();
+          // TODO: should be able to sign?
+          await expect(tx.sign()).rejects.toThrow(InvalidDataReason.LEDGER_POLICY);
+        });
+
+        it('would throw while trying to sign a transaction with ResignCommitteeCold certs', async () => {
+          const credential = getStakeCredential(
+            Cardano.RewardAccount('stake_test1up7pvfq8zn4quy45r2g572290p9vf99mr9tn7r9xrgy2l2qdsf58d')
+          );
+          const txBuilder = wallet.createTxBuilder();
+          txBuilder.partialTxBody.certificates = [
+            {
+              __typename: Cardano.CertificateType.ResignCommitteeCold,
+              anchor: null,
+              coldCredential: credential
+            }
+          ];
+          const tx = txBuilder
+            .addOutput(
+              txBuilder.buildOutput().address(outputs[0].address).coin(BigInt(outputs[0].value.coins)).toTxOut()
+            )
+            .build();
+
+          await expect(tx.sign()).rejects.toThrow(
+            InvalidDataReason.SIGN_MODE_ORDINARY__CERTIFICATE_COMMITTEE_COLD_CREDENTIAL_ONLY_AS_PATH
+          );
+        });
+
+        it('can sign a transaction with RegisterDelegateRepresentative certs', async () => {
+          const credential = getStakeCredential((await firstValueFrom(wallet.delegation.rewardAccounts$))?.[0].address);
+          const txBuilder = wallet.createTxBuilder();
+          txBuilder.partialTxBody.certificates = [
+            {
+              __typename: Cardano.CertificateType.RegisterDelegateRepresentative,
+              anchor: null,
+              dRepCredential: credential,
+              deposit: 5n
+            }
+          ];
+          const tx = txBuilder
+            .addOutput(
+              txBuilder.buildOutput().address(outputs[0].address).coin(BigInt(outputs[0].value.coins)).toTxOut()
+            )
+            .build();
+
+          expect(await tx.sign()).toBeTruthy();
+          // TODO: should be able to sign?
+          // await expect(tx.sign()).rejects.toThrow(InvalidDataReason.LEDGER_POLICY);
+        });
+
+        it.todo('would throw while trying to sign a transaction with RegisterDelegateRepresentative certs');
+
+        it.todo('can sign a transaction with UnregisterDelegateRepresentative certs');
+
+        it.todo('would throw while trying to sign a transaction with UnregisterDelegateRepresentative certs');
+
+        it.todo('can sign a transaction with UpdateDelegateRepresentative certs');
+
+        it.todo('would throw while trying to sign a transaction with UpdateDelegateRepresentative certs');
```

Comment:
@mchappell do i need to test `MULTISIG_TRANSACTION` mode as well?

Reply:
@AngelCastilloB could you please suggest how do i build proper multisig tx via txBuilder?

Reply:
@vetalcore I am not sure if the txBuilder supports multisig transaction currently, but you can see an example using the wallet directly here https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/e2e/test/wallet/PersonalWallet/multisignature.test.ts

Reply:
would address multisig in a separate ticket